### ISSUE TITLE
Add Siamese Network example

### ISSUE THREAD

**Issue by piyush-kgp** [pytorch/examples#645]
Hi, I  want to add an example for Siamese network, since it is one of the popular use cases in ML. I am thinking of implementing it in a way similar to other examples viz. command line arguments to choose which dataset to train, hyperparameters etc.
Is there something I need to keep in mind specifically apart from these:
- Use torchvision's Dataset class and PyTorch's DataLoader class to handle data.
- Implement a simple CNN as a nn.Module subclass
- Implement triplet loss
- Create train and test functions and a main function that calls those 2 methods at each epoch.
- Report final accuracy
- Maybe also report mAP score at the end.

Is this something that is worth adding to the repository.

**Comment by msaroufim** [comment]
Hi @piyush01123 you're welcome to try producing an example, I'd suggest taking a look at existing examples to see how to get going

**Comment by sudomaze** [comment]
I have implemented Siamese Network example in this PR https://github.com/pytorch/examples/pull/1003

**Comment by sudomaze** [comment]
I will close this issue as #1003 has been merged to main. Thanks, @msaroufim for the feedback!
